cleanlab/rank.py

```diff
@@ -109,57 +123,335 @@ def get_self_confidence_for_each_label(
+      .. seealso::
+        :func:`self_confidence`
+        :func:`normalized_margin`
+        :func:`confidence_weighted_entropy`
+
+    adj_pred_probs : bool, default = True
+      Adj_pred_probs in the same format expected by the `score_label_quality()` method.
+
+    weight_ensemble_members_by : {"uniform", "accuracy"}, default="uniform"
+      Weighting scheme used to aggregate scores from each model:
+        "uniform": Take the simple average of scores
+        "accuracy": Take weighted average of scores, weighted by model accuracy
+
+    verbose : int
+      If 0, no print statements. If 1, print weighting scheme and accuracy of each model.
+
     Returns
     -------
-      self_confidence : np.array (float)
-        Return the holdout probability that each example in pred_probs belongs to its
-        label. Assumes pred_probs is computed holdout/out-of-sample."""
+    label_quality_scores : np.array (float)
+
+    See Also
+    --------
+    self_confidence
+    normalized_margin
+    confidence_weighted_entropy
+    score_label_quality
+
+    """
+
+    # Check pred_probs_list for errors
+    assert isinstance(
+        pred_probs_list, list
+    ), f"pred_probs_list needs to be a list. Provided pred_probs_list is a {type(pred_probs_list)}"
+
+    assert len(pred_probs_list) > 0, "pred_probs_list is empty."
+
+    if len(pred_probs_list) == 1:
+        warnings.warn(
+            """
+            pred_probs_list only has one element. 
+            Consider using score_label_quality() if you only have a single array of pred_probs.
+            """
+        )
+
+    # Generate scores for each model's pred_probs
+    scores_list = []
+    accuracy_list = []
+    for pred_probs in pred_probs_list:
+
+        # Calculate scores and accuracy
+        scores = score_label_quality(
+            labels=labels,
+            pred_probs=pred_probs,
+            method=method,
+            adj_pred_probs=adj_pred_probs,
+        )
+        accuracy = (pred_probs.argmax(axis=1) == labels).mean()
+        scores_list.append(scores)
+        accuracy_list.append(accuracy)
+
+    # Print statements if enabled by verbose
+    if verbose:
+        print(f"Weighting scheme for ensemble: {weight_ensemble_members_by}")
+
+        # Only print if the user specified to weight by accuracy
+        if weight_ensemble_members_by == "accuracy":
+            for i, acc in enumerate(accuracy_list):
+                print(f"Model {i} accuracy: {acc}")
+
+    # Transform list of scores into an array of shape (N, M) where M is the number of models in the ensemble
+    scores_ensemble = np.vstack(scores_list).T
+
+    # Aggregate scores with chosen weighting scheme
+    if weight_ensemble_members_by == "uniform":
+
+        # Uniform weights (simple average)
+        label_quality_scores = scores_ensemble.mean(axis=1)
+
+    elif weight_ensemble_members_by == "accuracy":
+
+        # Weight by accuracy
+        weights = np.array(accuracy_list) / sum(accuracy_list)
```

Comment:
Let's also print out the aggregation weights that will be used when `verbosity > 0`

Reply:
Done 3dbbe7054aeae25b8bddd73a53838cce35b06fa9